internal/declarative/v2/reconciler.go

```diff
@@ -231,12 +239,24 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return r.finishReconcile(ctx, manifest, metrics.ManifestReconcileFinished, manifestStatus, nil)
 }
 
+func (r *Reconciler) handleLabelsRemovalFinalizerForUnmanagedModule(ctx context.Context,
```

Comment:
Can you extract this into a separat usecase / service?
This will lead to also needing to extract some kind of `ManifestApiClient` interface which will provide the manifest updating via:
`PatchStatusIfDiffExist`
`SsaSpec`
and `UpdateManifest`
which are also used by the Reconciler itself. Basically bundling the controller-runtime interfaces of `StatusClient` and `Writer`.
Sorry, this will result in some kind of extra work, but we need to start extracting parts of this monolithic Reconciler.
This would enable us to write unit tests for this use case and also move the `EventRecorder` dep from Reconciler to it.